Make docker daemon and init script agree on pidfile location

PaulFurtado opened the pull request:
**\- What I did**
Made docker daemon and init script agree on pidfile location

**\- How I did it**
Since the docker daemon has been renamed to dockerd, the init script now looks for
the pid file at /var/run/dockerd.pid, however, the docker daemon still writes the
pid file to /var/run/docker.pid by default so the init script thinks it fails to
start and cannot stop the service either.

This change makes the init script pass the --pidfile option to the docker daemon,
so the daemon and init script will always agree on the pid file location.

**\- How to verify it**
1. Build and install version 1.12.0-rc2 of the docker RPM for CentOS 6. (Also configure the prereqs for CentOS 6 installation: update kernel, iptables, etc)
2. Attempt to start the service, note that it times out and fails, but the service started successfully
3. Apply this patch, repeat previous steps, notice that the service starts and stops successfully.

**\- Description for the changelog**
Made docker daemon and init script agree on pidfile location now that the name of the daemon is dockerd.

Signed-off-by: Paul Furtado paulfurtado91@gmail.com

thaJeztah commented:
/cc @mlaventure @tonistiigi

mlaventure commented:
LGTM (IANAM)

thaJeztah commented:
added this to the 1.12 milestone as it's related to the introduction of `dockerd`

vdemeester commented:
LGTM 🐯
/cc @runcom too

runcom commented:
unless I miss something, why not just renaming the `.pid` file here https://github.com/docker/docker/pull/24147/files#diff-ddea2a9ee81bcf853a0b8a85c5fa93e8R28 to be `/var/run/docker.pid` directly? I believe there can't be `dockerd.pid` (notice the d) so the line broken is https://github.com/docker/docker/pull/24147/files#diff-ddea2a9ee81bcf853a0b8a85c5fa93e8R28 which needs changing

mlaventure commented:
This need a rebase

PaulFurtado commented:
@runcom Yes, the thing that was broken was that docker still wrote its pid file to `docker.pid`, but the init script had been updated to look for it at `dockerd.pid`. I think you and I are on the same page here. I figured you guys actually *wanted* to rename everything to dockerd, so I made this PR this way, but it's definitely not what I prefer because it breaks backwards compatibility (requires people to start using `/etc/sysconfig/dockerd` instead of `/etc/sysconfig/docker` and calling `service docker restart` after upgrading from 1.11 to 1.12 will not successfully stop 1.11 because it looks at the wrong pid file).

What we're actually using in our infrastructure right now is this commit here: https://github.com/PaulFurtado/docker/commit/c91ad1715d1005c11fa64b6ba37717c4dd12e9f1

There, I did as you suggested here and did prog=docker and set the exe to dockerd. Way better user experience.

I'll squash that change down to a single commit, rebase onto master, and update this PR.

PaulFurtado committed:
Change $prog back to docker in sysvinit-redhat

(and set $exec to dockerd instead)

This ensures end users do not need to make any configuration changes
due to the rename from docker to dockerd in version 1.12.

Signed-off-by: Paul Furtado <pfurtado@hubspot.com>

GordonTheTurtle commented:
Please sign your commits following these rules:
https://github.com/docker/docker/blob/master/CONTRIBUTING.md#sign-your-work
The easiest way to do this is to amend the last commit:
~~~console
$ git clone -b "fix_init_pidfile_dockerd" git@github.com:PaulFurtado/docker.git somewhere
$ cd somewhere
$ git commit --amend -s --no-edit
$ git push -f
~~~

Ammending updates the existing PR. You **DO NOT** need to open a new one.

runcom commented:
mmm still not sure about this @PaulFurtado - I though just leaving the `.pid` file as docker should do the trick, not sure about the other files, I remember with @tianon we said that those should be changed as well to dockerd - so maybe the task in this PR is to just set the pid file to `docker.pid` - not chaning `$prog` to `docker` which will cause all other files to be renamed back to docker which isn't right to me.

ping @tianon could you shed some light here? When we made this changes we probably mistaken changing the pid file to `dockerd.pid` because we renamed that `$prog` var and it affected the pid file name which wasn't right. So, I think that we only need to set the pid file back to `docker.pid` not the other files like `/etc/sysconfig/dockerd/` but I can be wrong so I'll let you talk here :)

tianon commented:
Yeah, I think @runcom is correct -- we do still want `prog` to be `dockerd`, but the PID file specifically should still be `docker.pid`. :+1:

PaulFurtado commented:
So, changing `$prog` to `"dockerd"` will result in the settings file changing from `/etc/sysconfig/docker` to `/etc/sysconfig/dockerd`. Is that actually desirable?

tianon commented:
Oh hmm, good point.  I think in this case you're probably right -- I believe it's the case of `$DOCKERD` vs `$DOCKER` that @runcom and I are getting confused about (where we _do_ want the breaking change of the variable becoming `DOCKERD`, but that doesn't apply to this particular script).

tianon commented:
LGTM

icecrime commented:
@vdemeester PTAL!

runcom commented:
@tianon correct, that what I was thinking as well 

LGTM

vdemeester commented:
LGTM 🐸
